barman/diagnose.py

```diff
@@ -80,6 +83,19 @@ def exec_diagnose(servers, errors_list):
         diagnosis["servers"][name]["status"] = server.get_remote_status()
         # backup list
         backups = server.get_available_backups(BackupInfo.STATUS_ALL)
+        # update date format for each backup begin_time and end_time and ensure local timezone.
+        # This code is a duplicate from BackupInfo.to_json()
+        # This should be temporary to keep original behavior for other usage.
```

Comment:
Changes in diagnose.py file are really ugly, but it was the easiest way to not impact other `BackupInfo.to_json()`  users.

If we plan to use isoformat() for all `BackupInfo.to_json()`  users then it is a temporary thing. Otherwise, we should consider finding another solution.

Reply:
Ideally this would be a temporary thing - we can file an issue for replacing ctime strings with ISO 8601 format strings in all the output and that could land in a major version change, which would allow us to remove this code and probably revert back to a single Barman encoder.

Reply:
I like this idea.
Hopping no one will miss ctime output.